src/core/namespace.c

```diff
@@ -3310,6 +3313,379 @@ bool ns_type_supported(NamespaceType type) {
+                _cleanup_(sigkill_waitp) pid_t grandchild_pid = 0;
+
+                 (void) mkdir_p_label(overlay_prefix, 0555);
+
+                /* Open all extension roots on the host */
+                FOREACH_ARRAY(m, ml.mounts, ml.n_mounts)
+                        if (IN_SET(m->mode, MOUNT_EXTENSION_DIRECTORY, MOUNT_EXTENSION_IMAGE)) {
+                                r = apply_one_mount("/", m, p);
+                                if (r < 0) {
+                                        log_debug_errno(r, "Failed to apply extension mount: %m");
+                                        _exit(EXIT_FAILURE);
+                                }
+                        }
+
+                /* Create a grandchild process to handle the unmounting and reopening of hierarchy */
+                r = socketpair(AF_UNIX, SOCK_DGRAM|SOCK_CLOEXEC, 0, pair);
+                if (r < 0) {
+                        log_debug_errno(errno, "Failed to create socket pair: %m");
+                        _exit(EXIT_FAILURE);
+                }
+
+                r = safe_fork("(sd-ns-refresh-exts-grandchild)",
+                                FORK_LOG|FORK_DEATHSIG_SIGKILL,
+                                &grandchild_pid);
+                if (r < 0)
+                        _exit(EXIT_FAILURE);
+                if (r == 0) {
+                        /* Grandchild (target service namespace) */
+                        pair[0] = safe_close(pair[0]);
+
+                        r = pidref_namespace_open(target, &pidns_fd, &mntns_fd, NULL, NULL, &root_fd);
+                        if (r < 0) {
+                                log_debug_errno(r, "Failed to open namespace: %m");
+                                _exit(EXIT_FAILURE);
+                        }
+
+                        r = is_our_namespace(mntns_fd, NAMESPACE_MOUNT);
+                        if (r < 0) {
+                                log_debug_errno(r, "Failed to check if target namespace is separate: %m");
+                                _exit(EXIT_FAILURE);
+                        }
+                        if (r > 0) {
+                                log_debug("Target namespace is not separate, cannot reload extensions");
+                                _exit(EXIT_FAILURE);
+                        }
+
+                        r = namespace_enter(pidns_fd, mntns_fd, -EBADF, -EBADF, root_fd);
+                        if (r < 0) {
+                                log_debug_errno(r, "Failed to enter namespace: %m");
+                                _exit(EXIT_FAILURE);
+                        }
+
+                        /* Handle each overlay mount path */
+                        FOREACH_ARRAY(m, ml.mounts, ml.n_mounts) {
+                                if (m->mode != MOUNT_OVERLAY)
+                                        continue;
+
+                                const char *mount_path = startswith(mount_entry_unprefixed_path(m), overlay_prefix);
```

Comment:
path_startswith()

Reply:
this drops the leading `/` which breaks using it, ie, you get `etc` instead of `/etc`